fastlane/lib/fastlane/actions/device_grid/device_grid.rb

```diff
@@ -56,6 +58,8 @@ def run(public_key: nil, languages: nil, devices: nil, prefix_command: nil)
             markdown("</tr>")
           end
           markdown("</table>")
+
+          ENV.delete(fastlane_colors_env) unless fastlane_colors_were_disabled
```

Comment:
should that be under an `ensure` ? And only if `fastlane_colors_were_disabled` was set.